examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -0,0 +1,298 @@
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrReturnError(IsSpanUsable(out_signature_buffer), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(IsSpanUsable(message_to_sign), CHIP_ERROR_INVALID_ARGUMENT);
+
+    ChipLogDetail(Crypto, "Sign using DA key from se05x (Using internal sign)");
+
+    TLV::TLVReader msg_reader;
+    TLV::TLVReader tagReader;
+
+    msg_reader.Init(message_to_sign);
+
+    /* To be removed. Use common key id to sign message */
+    static bool sign_cert_decl_attest = 1;
+
+    if (sign_cert_decl_attest)
+    {
+        /* Check if certificate declaration tag is present and Skip certificate declaration tag */
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
+        uint8_t attlen = tagReader.GetLength();
+        VerifyOrReturnError(attlen > 0, CHIP_ERROR_INVALID_TLV_TAG);
+        /* Get attestation nonce */
+        ByteSpan attest_nonce;
+        ReturnErrorOnFailure(tagReader.Get(attest_nonce));
+        /* Set attestation nonce */
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(3), tagReader));
+        uint8_t tslen = tagReader.GetLength();
+        if (tslen > 0)
+        {
+            ByteSpan time_stamp;
+            ReturnErrorOnFailure(tagReader.Get(time_stamp));
+            /* Set time stamp data */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_TIME_STAMP_DATA_SE05X_ID, time_stamp.data(), time_stamp.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+        /* Set time stamp length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tslen, 1), CHIP_ERROR_INTERNAL);
+
+        if ((tagReader.GetRemainingLength() + 1 /* End container */) >= 16)
+        {
+            /* Set attestation challenge */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_ATTEST_CHALLENGE_SE05X_ID, (message_to_sign.end() - 16), 16),
+                                CHIP_ERROR_INTERNAL);
+        }
+    }
+    else
+    {
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+        uint8_t csrlen = tagReader.GetLength();
+        VerifyOrReturnError(csrlen > 0, CHIP_ERROR_INVALID_TLV_TAG);
+        ByteSpan csr_data;
+        /* Get nocsr */
+        ReturnErrorOnFailure(tagReader.Get(csr_data));
+        /* Set nocsr length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_LEN_SE05X_ID, &csrlen, 1), CHIP_ERROR_INTERNAL);
```

Comment:
Why not initializing csrlen from `csr_data.size()`?

Reply:
get() member will return error in case the tag length is 0. So using the GetLength() before get() member is called.

Reply:
Again, `Get` does not return error on zero length....

Reply:
I have created the issue to check this as well as other changes you suggested in the previous conversation. (https://github.com/project-chip/connectedhomeip/issues/23064). Will address these in the next PR.